c10/core/TensorImpl.cpp

```diff
@@ -95,6 +98,11 @@ TensorImpl::TensorImpl(Storage&& storage, DispatchKeySet key_set, const caffe2::
     key_set_ = key_set | getAutogradRelatedKeySetFromBackend(k);
   }
 
+  // Inference tensor doesn't have version counter.
+  if (!is_inference_tensor()) {
+    version_counter_ = VariableVersion();
```

Comment:
version_counter_(VariableVersion(VariableVersion::DISABLED)) does nothing ;) it's here since our default constructor does create intrusive ptr so we want to avoid.

Reply:
Well it does create the wrapper object (even though it doesn't create the struct containing the atomic).